jbpm-case-mgmt/jbpm-case-mgmt-impl/src/main/java/org/jbpm/casemgmt/impl/CaseServiceImpl.java

```diff
@@ -438,45 +428,62 @@ public void removeFromCaseRole(String caseId, String role, OrganizationalEntity
     public Collection<CaseRoleInstance> getCaseRoleAssignments(String caseId) throws CaseNotFoundException {
         authorizationManager.checkOperationAuthorization(caseId, ProtectedOperation.MODIFY_ROLE_ASSIGNMENT);
         ProcessInstanceDesc pi = verifyCaseIdExists(caseId);
-        
+
         CaseFileInstance caseFile = internalGetCaseFileInstance(caseId, pi.getDeploymentId());
-        
-        return ((CaseFileInstanceImpl)caseFile).getAssignments();
+
+        return ((CaseFileInstanceImpl) caseFile).getAssignments();
     }
 
-    
     /*
      * Case comments methods
      */
-    
 
     @Override
     public Collection<CommentInstance> getCaseComments(String caseId, QueryContext queryContext) throws CaseNotFoundException {
         authorizationManager.checkOperationAuthorization(caseId, ProtectedOperation.MODIFY_COMMENT);
         ProcessInstanceDesc pi = verifyCaseIdExists(caseId);
-        
+
         CaseFileInstance caseFile = internalGetCaseFileInstance(caseId, pi.getDeploymentId());
-        
-        return ((CaseFileInstanceImpl)caseFile).getComments();
+
+        return ((CaseFileInstanceImpl) caseFile).getComments();
     }
 
     @Override
     public Collection<CommentInstance> getCaseComments(String caseId, CommentSortBy sortBy, QueryContext queryContext) throws CaseNotFoundException {
         authorizationManager.checkAuthorization(caseId);
         Collection<CommentInstance> comments = getCaseComments(caseId, queryContext);
-        
+
         return comments.stream().sorted((o1, o2) -> {
             int result = 0;
             switch (sortBy) {
-                case Date : 
+                case Date:
                     result = o1.getCreatedAt().compareTo(o2.getCreatedAt());
                     break;
-                case Author : 
+                case Author:
                     result = o1.getAuthor().compareTo(o2.getAuthor());
                     break;
             }
             return result;
-        }).collect(Collectors.toList());          
+        }).collect(Collectors.toList());
+    }
+
+    @Override
+    public Collection<CommentInstance> getCaseComments(String caseId, int page, int pageSize, QueryContext queryContext) throws CaseNotFoundException {
+        authorizationManager.checkAuthorization(caseId);
+        Collection<CommentInstance> comments = getCaseComments(caseId, queryContext);
+
+        List<CommentInstance> commentsToPaginate = new ArrayList<>(comments);
+
+        int allCommentsSize = commentsToPaginate.size();
+        int pageIndex = (allCommentsSize + pageSize - 1) / pageSize;
+
+        if (allCommentsSize < pageSize) {
+            return comments;
+        } else if (pageIndex == page + 1) {
+            return commentsToPaginate.subList(page * pageSize, allCommentsSize);
+        } else {
+            return commentsToPaginate.subList(page * pageSize, (page * pageSize) + pageSize);
```

Comment:
@mswiderski @annabaker I would still be in favor of adding the pagination here, to try to keep consistency in how the UI interact with most services, limiting the amount of data we transfer and load. That I think can also be future proof in terms of core services changing how comments are stored/loaded for instance.